src/ui/components/Tab.tsx

```diff
@@ -1,42 +1,43 @@
 import type { editor as Editor } from 'monaco-editor';
 import type { FunctionalComponent } from 'preact';
-import { h, Fragment } from 'preact';
+import { h } from 'preact';
 import path from 'path';
 
 const noop = () => {};
 
 export interface Props {
+  'aria-selected': boolean;
   index: number;
   model: Editor.IModel;
-  "aria-selected"?: boolean;
   onRemoveTab?: (model: Editor.IModel) => void;
   onSetTab?: (model: Editor.IModel) => void;
 }
-const Tab: FunctionalComponent<Props> = ({ "aria-selected": active = false, index, onRemoveTab = noop, onSetTab = noop, model }) => {
-    return (
-        <li class={`ap-preview-tabwrapper ${index !== 0 ? 'ap-tab-can-remove' : ''}`.trim()}>
-            <button
-              role="tab"
-              type="button"
-              aria-selected={active}
-              class="ap-preview-tab"
-              onClick={() => onSetTab(model)}
-            >
-              {path.posix.basename(model.uri.path)}
-            </button>
-            {index !== 0 && (
-              <button
-                type="button"
-                class="ap-tab-remove"
-                onClick={() => onRemoveTab(model)}
-              >
-                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
-                  <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16" />
-                </svg>
-              </button>
-            )}
-        </li>
-    )
-}
+const Tab: FunctionalComponent<Props> = ({ 'aria-selected': active = false, index, onRemoveTab = noop, onSetTab = noop, model }) => {
+  const tabName = path.posix.basename(model.uri.path);
+  return (
+    <div class={`ap-tabgroup-tabwrapper${index !== 0 ? ' is-removable' : ''}`}>
+      <button role="tab" type="button" aria-selected={active} aria-controls="editor" class="ap-tabgroup-tab" onClick={() => onSetTab(model)}>
+        {tabName}
+      </button>
+      {index !== 0 && (
+        <button
+          type="button"
+          class="ap-tabgroup-remove"
+          onClick={() => (confirm(`Delete ${tabName}? This can’t be undone.`) ? onRemoveTab(model) : {})}
```

Comment:
![Screen Shot 2021-09-13 at 20 57 15](https://user-images.githubusercontent.com/1369770/133187391-6064d222-8592-4eb1-bec3-940f897318be.png)

Adds a little `confirm()` dialog just in case someone was working really hard on something, tried to click the tab, and missed by a few px! 😅